### ISSUE TITLE
ValueError: 'from' is not a valid parameter name

### ISSUE THREAD

**Issue by chriselrod** [regen100/cmake-language-server#70]
```
[stderr]   File "/home/chriselrod/.local/lib/python3.11/site-packages/pydantic/main.py", line 369, in __new__
[stderr]     cls.__signature__ = ClassAttribute('__signature__', generate_model_signature(cls.__init__, fields, config))
[stderr]                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[stderr]   File "/home/chriselrod/.local/lib/python3.11/site-packages/pydantic/utils.py", line 231, in generate_model_signature
[stderr]     merged_params[param_name] = Parameter(
[stderr]                                 ^^^^^^^^^^
[stderr]   File "/usr/lib64/python3.11/inspect.py", line 2721, in __init__
[stderr]     raise ValueError('{!r} is not a valid parameter name'.format(name))
[stderr] ValueError: 'from' is not a valid parameter name
[internal] Fri Feb  3 08:23:50 2023:
(:message "Connection state changed" :change "exited abnormally with code 1\n")
```

I am on python 3.11 and cmake-language-server 0.1.5 (even though 0.1.6 seems to be the latest release) because `pip list --outdated` doesn't list `cmake-language-server` and an explicit upgrade claims it's already satisfied
```sh
$ pip install cmake-language-server -U
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: cmake-language-server in /home/chriselrod/.local/lib/python3.11/site-packages (0.1.5)
```

**Comment by comiluv** [comment]
I'm also getting similar error in the lsp log when I open CMakeLists.txt in Neovim where I installed this server with mason.nvim and configured using lsp-zero.nvim and nvim-lspconfig.
My error messages in lsp.log are as below:
```log
[START][2023-02-06 15:47:12] LSP logging initiated
[ERROR][2023-02-06 15:47:12] .../vim/lsp/rpc.lua:733	"rpc"	"C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\bin\\cmake-language-server.CMD"	"stderr"	'Traceback (most recent call last):\r\n  File "<frozen runpy>", line 198, in _run_module_as_main\r\n  File "<frozen runpy>", line 88, in _run_code\r\n'
[ERROR][2023-02-06 15:47:12] .../vim/lsp/rpc.lua:733	"rpc"	"C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\bin\\cmake-language-server.CMD"	"stderr"	'  File "C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\packages\\cmake-language-server\\venv\\Scripts\\cmake-language-server.exe\\__main__.py", line 4, in <module>\r\n  File "C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\packages\\cmake-language-server\\venv\\Lib\\site-packages\\cmake_language_server\\server.py", line 8, in <module>\r\n    from pygls.lsp.methods import (\r\n  File "C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\packages\\cmake-language-server\\venv\\Lib\\site-packages\\pygls\\lsp\\__init__.py", line 23, in <module>\r\n'
[ERROR][2023-02-06 15:47:12] .../vim/lsp/rpc.lua:733	"rpc"	"C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\bin\\cmake-language-server.CMD"	"stderr"	'    from pygls.lsp.types import *\r\n  File "C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\packages\\cmake-language-server\\venv\\Lib\\site-packages\\pygls\\lsp\\types\\__init__.py", line 6, in <module>\r\n    from pygls.lsp.types.general_messages import *\r\n'
[ERROR][2023-02-06 15:47:12] .../vim/lsp/rpc.lua:733	"rpc"	"C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\bin\\cmake-language-server.CMD"	"stderr"	'  File "C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\packages\\cmake-language-server\\venv\\Lib\\site-packages\\pygls\\lsp\\types\\general_messages.py", line 35, in <module>\r\n'
[ERROR][2023-02-06 15:47:12] .../vim/lsp/rpc.lua:733	"rpc"	"C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\bin\\cmake-language-server.CMD"	"stderr"	"    from pygls.lsp.types.language_features import (CallHierarchyClientCapabilities,\r\n"
[ERROR][2023-02-06 15:47:12] .../vim/lsp/rpc.lua:733	"rpc"	"C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\bin\\cmake-language-server.CMD"	"stderr"	'  File "C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\packages\\cmake-language-server\\venv\\Lib\\site-packages\\pygls\\lsp\\types\\language_features\\__init__.py", line 2, in <module>\r\n'
[ERROR][2023-02-06 15:47:12] .../vim/lsp/rpc.lua:733	"rpc"	"C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\bin\\cmake-language-server.CMD"	"stderr"	'    from pygls.lsp.types.language_features.call_hierarchy import *\r\n  File "C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\packages\\cmake-language-server\\venv\\Lib\\site-packages\\pygls\\lsp\\types\\language_features\\call_hierarchy.py", line 70, in <module>\r\n'
[ERROR][2023-02-06 15:47:12] .../vim/lsp/rpc.lua:733	"rpc"	"C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\bin\\cmake-language-server.CMD"	"stderr"	'    class CallHierarchyIncomingCall(Model):\r\n  File "C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\packages\\cmake-language-server\\venv\\Lib\\site-packages\\pydantic\\main.py", line 369, in __new__\r\n'
[ERROR][2023-02-06 15:47:12] .../vim/lsp/rpc.lua:733	"rpc"	"C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\bin\\cmake-language-server.CMD"	"stderr"	"    cls.__signature__ = ClassAttribute('__signature__', generate_model_signature(cls.__init__, fields, config))\r\n"
[ERROR][2023-02-06 15:47:12] .../vim/lsp/rpc.lua:733	"rpc"	"C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\bin\\cmake-language-server.CMD"	"stderr"	'                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\r\n  File "C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\packages\\cmake-language-server\\venv\\Lib\\site-packages\\pydantic\\utils.py", line 231, in generate_model_signature\r\n'
[ERROR][2023-02-06 15:47:12] .../vim/lsp/rpc.lua:733	"rpc"	"C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\bin\\cmake-language-server.CMD"	"stderr"	'    merged_params[param_name] = Parameter(\r\n                                ^^^^^^^^^^\r\n  File "C:\\Python311\\Lib\\inspect.py", line 2715, in __init__\r\n'
[ERROR][2023-02-06 15:47:12] .../vim/lsp/rpc.lua:733	"rpc"	"C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\bin\\cmake-language-server.CMD"	"stderr"	"    raise ValueError('{!r} is not a valid parameter name'.format(name))\r\n"
[ERROR][2023-02-06 15:47:12] .../vim/lsp/rpc.lua:733	"rpc"	"C:\\Users\\choij\\AppData\\Local\\nvim-data\\mason\\bin\\cmake-language-server.CMD"	"stderr"	"ValueError: 'from' is not a valid parameter name\r\n"
```

Windows 11 22H2
NVIM 0.8.3
cmake-language-server 0.1.5

**Comment by martinclauss** [comment]
Same here with Python 3.11.1 under Fedora 37...

**Comment by chriselrod** [comment]
Note
```
$ pip install cmake-language-server==0.1.6
Defaulting to user installation because normal site-packages is not writeable
ERROR: Ignored the following versions that require a different python version: 0.1.6 Requires-Python >=3.7,<3.11
ERROR: Could not find a version that satisfies the requirement cmake-language-server==0.1.6 (from versions: 0.1.0, 0.1.1, 0.1.2, 0.1.3, 0.1.4, 0.1.5)
ERROR: No matching distribution found for cmake-language-server==0.1.6
```
I'll try master tomorrow.

**Comment by breakersun** [comment]
Windows 10 22H2, Python 3.11.1

Is this just me?

```
cmake_test on  master 🛤️ ×1 via  △ v3.25.2
❌ pip install cmake-language-server -U                                                                          at 18:00:01
Requirement already satisfied: cmake-language-server in c:\users\403283\scoop\apps\python\current\lib\site-packages (0.1.5)
Requirement already satisfied: pygls<0.12,>=0.11 in c:\users\403283\scoop\apps\python\current\lib\site-packages (from cmake-language-server) (0.11.3)
Requirement already satisfied: pydantic<1.9,>=1.7 in c:\users\403283\scoop\apps\python\current\lib\site-packages (from pygls<0.12,>=0.11->cmake-language-server) (1.8.2)
Requirement already satisfied: typeguard<3,>=2.10.0 in c:\users\403283\scoop\apps\python\current\lib\site-packages (from pygls<0.12,>=0.11->cmake-language-server) (2.13.3)
Requirement already satisfied: typing-extensions>=3.7.4.3 in c:\users\403283\scoop\apps\python\current\lib\site-packages (from pydantic<1.9,>=1.7->pygls<0.12,>=0.11->cmake-language-server) (4.5.0)

[notice] A new release of pip available: 22.3.1 -> 23.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip

cmake_test on  master 🛤️ ×1 via  △ v3.25.2
❯ cmake-language-server.exe --version                                                                            at 18:01:15
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\403283\scoop\apps\python\current\Scripts\cmake-language-server.exe\__main__.py", line 4, in <module>
  File "C:\Users\403283\scoop\apps\python\current\Lib\site-packages\cmake_language_server\server.py", line 8, in <module>
    from pygls.lsp.methods import (
  File "C:\Users\403283\scoop\apps\python\current\Lib\site-packages\pygls\lsp\__init__.py", line 23, in <module>
    from pygls.lsp.types import *
  File "C:\Users\403283\scoop\apps\python\current\Lib\site-packages\pygls\lsp\types\__init__.py", line 6, in <module>
    from pygls.lsp.types.general_messages import *
  File "C:\Users\403283\scoop\apps\python\current\Lib\site-packages\pygls\lsp\types\general_messages.py", line 35, in <module>
    from pygls.lsp.types.language_features import (CallHierarchyClientCapabilities,
  File "C:\Users\403283\scoop\apps\python\current\Lib\site-packages\pygls\lsp\types\language_features\__init__.py", line 2, in <module>
    from pygls.lsp.types.language_features.call_hierarchy import *
  File "C:\Users\403283\scoop\apps\python\current\Lib\site-packages\pygls\lsp\types\language_features\call_hierarchy.py", line 70, in <module>
    class CallHierarchyIncomingCall(Model):
  File "C:\Users\403283\scoop\apps\python\current\Lib\site-packages\pydantic\main.py", line 369, in __new__
    cls.__signature__ = ClassAttribute('__signature__', generate_model_signature(cls.__init__, fields, config))
                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\403283\scoop\apps\python\current\Lib\site-packages\pydantic\utils.py", line 231, in generate_model_signature
    merged_params[param_name] = Parameter(
                                ^^^^^^^^^^
  File "C:\Users\403283\scoop\apps\python\current\Lib\inspect.py", line 2721, in __init__
    raise ValueError('{!r} is not a valid parameter name'.format(name))
ValueError: 'from' is not a valid parameter name
```

**Comment by chriselrod** [comment]
I haven't tried it much, but I at least don't get immediate errors when installing via
```sh
pip install git+https://github.com/regen100/cmake-language-server.git@master
```

**Comment by breakersun** [comment]
> I haven't tried it much, but I at least don't get immediate errors when installing via
> 
> ```shell
> pip install git+https://github.com/regen100/cmake-language-server.git@master
> ```

Thanks this works.

I was using mason lsp server manager inside neovim, it was using the latest release instead of main branch, so this problem remains inside neovim + mason.

**Comment by regen100** [comment]
Version 0.1.7 has been released with support for Python 3.11 and the new pygls. If you have problems again, please reopen this issue or create a new one.
